site/src/pages/WorkspaceSettingsPage/WorkspaceSharingPage/WorkspaceSharingPage.tsx

```diff
@@ -1,19 +1,170 @@
-import { PageHeader, PageHeaderTitle } from "components/PageHeader/PageHeader";
+import { getErrorMessage } from "api/errors";
+import { checkAuthorization } from "api/queries/authCheck";
+import {
+	setWorkspaceGroupRole,
+	setWorkspaceUserRole,
+	workspaceACL,
+} from "api/queries/workspaces";
+import { ErrorAlert } from "components/Alert/ErrorAlert";
+import { displayError, displaySuccess } from "components/GlobalSnackbar/utils";
+import type { WorkspacePermissions } from "modules/workspaces/permissions";
+import { workspaceChecks } from "modules/workspaces/permissions";
 import type { FC } from "react";
+import { useMutation, useQuery, useQueryClient } from "react-query";
 import { pageTitle } from "utils/page";
 import { useWorkspaceSettings } from "../WorkspaceSettingsLayout";
+import { WorkspaceSharingPageView } from "./WorkspaceSharingPageView";
 
 const WorkspaceSharingPage: FC = () => {
 	const workspace = useWorkspaceSettings();
+	const queryClient = useQueryClient();
+
+	const workspaceACLQuery = useQuery(workspaceACL(workspace.id));
+	const checks = workspaceChecks(workspace);
+	const permissionsQuery = useQuery({
+		...checkAuthorization({ checks }),
+	});
+	const permissions = permissionsQuery.data as WorkspacePermissions | undefined;
+
+	const addUserMutation = useMutation(setWorkspaceUserRole(queryClient));
+	const updateUserMutation = useMutation(setWorkspaceUserRole(queryClient));
+	const removeUserMutation = useMutation(setWorkspaceUserRole(queryClient));
+
+	const addGroupMutation = useMutation(setWorkspaceGroupRole(queryClient));
+	const updateGroupMutation = useMutation(setWorkspaceGroupRole(queryClient));
+	const removeGroupMutation = useMutation(setWorkspaceGroupRole(queryClient));
+
+	const canUpdatePermissions = Boolean(permissions?.updateWorkspace);
 
 	return (
-		<>
+		<div className="flex flex-col gap-12 max-w-screen-md">
 			<title>{pageTitle(workspace.name, "Sharing")}</title>
 
-			<PageHeader className="pt-0">
-				<PageHeaderTitle>Sharing</PageHeaderTitle>
-			</PageHeader>
-		</>
+			<header className="flex flex-col">
+				<div className="flex flex-col gap-2">
+					<h1 className="text-3xl m-0">Workspace sharing</h1>
+					<p className="flex flex-row gap-1 text-sm text-content-secondary font-medium m-0">
+						Workspace sharing allows you to share workspaces with other users
+						and groups.
+						{/* TODO: ADD DOCS LINK HERE <Link href={docs("/admin/users/idp-sync")}>View docs</Link> */}
```

Comment:
just fyi, the current url proposed by jiachen is `docs("/user-guides/workspaces-shared")` but I proposed some alternatives that I think are a bit better

I imagine that pr will probably be merged soon one way or another

https://github.com/coder/coder/pull/21214/changes#r2611975329